app/src/main/java/org/mozilla/fenix/home/recentvisits/view/RecentlyVisited.kt

```diff
@@ -153,13 +164,15 @@ private fun RecentlyVisitedHistoryGroup(
         ) {
             RecentlyVisitedTitle(
                 text = recentVisit.title,
-                modifier = Modifier.padding(top = 7.dp, bottom = 2.dp)
+                modifier = Modifier
+                    .padding(top = 7.dp, bottom = 2.dp)
+                    .weight(1f)
             )
 
-            RecentlyVisitedCaption(recentVisit.historyMetadata.size)
+            RecentlyVisitedCaption(recentVisit.historyMetadata.size, Modifier.weight(1f))
 
             if (showDividerLine) {
-                RecentlyVisitedDivider(modifier = Modifier.padding(top = 9.dp))
```

Comment:
Am I correct to assume that the weight fills in the 'remaining' space so that the Row is still 56dp with the divider being at the bottom?

Reply:
Yes.
Based on [the documentation](https://developer.android.com/reference/kotlin/androidx/compose/foundation/layout/ColumnScope?hl=pt#(androidx.compose.ui.Modifier).weight(kotlin.Float,kotlin.Boolean)) using weights will mean the following succession of events:
- first the unweighted elements are measured - in our case the divider will have the exact height that it needs
- then the remaining space (column height minus unweighted items height) is divided between weighted elements based on their weight - in our case the title and caption having the same weight will each have half of the remaining height
- then the items are laid out based on their measurements - in our case all items have perfect measurements, the title and caption will be placed on screen (with a bit bigger height than needed but the default Top/Start alignment plus the padding set for title gets the placement we want) then the divider as the last item in the column is guaranteed to be shown at the bottom and occupy the exact space it needs.

This approach helps ensuring all items will have the space they need on screen and more importantly by measuring the title and caption after the divider helps ensure the divider will always be shown on screen, at the bottom, while I think previously there were cases in which it would be pushed to the outside of the column.